### ISSUE TITLE
Union type shouldn't be assignable to constrained TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1345]
```python
from typing import AnyStr, Union, Sequence

def g(a: Sequence[AnyStr]) -> Sequence[AnyStr]:
    ...

def f(a: Sequence[Union[str, bytes]]):
    g(a)  # Should generate an error
```

Likewise

```python
from typing import AnyStr, Union, List

def g(a: AnyStr, b: AnyStr) -> None:
    ...

def f(a: Union[str, bytes], b: Union[str, bytes]):
    g(a, b)  # Should generate an error
```

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
